File: 30 Day Coding Challenge/2 October 2023.py

from sys import stdin

def findTriplet(arr, n, x) :
    #Your code goes here
    #return your answer
    count =0
    for i in range(n):
        for j in range(i+1,n):
            for k in range(j+1,n):
                if arr[i] + arr[k] + arr[j]==x:
                    count +=1
    
    return count

File: 30 Day Coding Challenge/2 October 2023.java
// Question 1
public class Solution {
    
    public static int findTriplet(int[] arr, int x) {
    	//Your code goes here
        int count =0;

        for(int i=0;i<arr.length;i++){
            for(int j=i+1;j<arr.length;j++){
                for(int k=j+1;k<arr.length;k++){
                    if(arr[i] + arr[j] + arr[k] ==x)    
                        count++;
                }
            }
        }

        return count;
    }

}


// Question 2

public class Solution {  

    public static void sortZeroesAndOne(int[] arr) {
    	//Your code goes here
        int j =arr.length-1,k=0;

        while(k<j){
            if(arr[k]==0){
                k++;
            } else {
                if(arr[j]==0){
                    arr[k]=0;
                    arr[j]=1;
                    k++;
                } 
                j--;
            }
        }
    }
}
